### PR TITLE
Feature/docs/101 reporting documentation (#101)

### PR DESCRIPTION
**Pull request by lucamrgs**
documentation-specific pull request for reporting functionalities.
Integrated @RabbITCybErSeC's feedback.

### CODE REVIEW

File: docs/content/en/docs/core-components/reporting.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+The logic and extensibility is implemented in the SOARCA architecture by means of reporting interfaces. At this stage, we implement an *IWorkflowReporter* to push information about the entire workflow to be executed, and an *IStepReporter* to push step-specific information as the steps of the workflow are executed.
+
+A high level *Reporter* class will implement both interfaces, and maintain the list of decomposer and executor reporters activated for the SOARCA instance. The *Reporter* class will invoke all reporting functions for each active reporter.
+
+## Future plans
+
+At this stage, third party tools integrations may be built in SOARCA via packages implementing reporting logic for the specific tools. Alternatively, third party tools may implement pull-based mechanisms to get information from the execution of a playbook via SOARCA.
+
+In the near future, we will (also) make available a SOARCA Report API that can establish a WebSocket connection to a third party tool. As such, this will thus allow SOARCA to push execution updates as they come to third party tools, without external tools having to poll SOARCA.
```
Comment (RabbITCybErSeC) [inline docs/content/en/docs/core-components/reporting.md:80]: again `third-party tools`

File: docs/content/en/docs/core-components/reporting.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+### Interfaces
+
+The logic and extensibility is implemented in the SOARCA architecture by means of reporting interfaces. At this stage, we implement an *IWorkflowReporter* to push information about the entire workflow to be executed, and an *IStepReporter* to push step-specific information as the steps of the workflow are executed.
+
+A high level *Reporter* class will implement both interfaces, and maintain the list of decomposer and executor reporters activated for the SOARCA instance. The *Reporter* class will invoke all reporting functions for each active reporter.
+
+## Future plans
+
+At this stage, third party tools integrations may be built in SOARCA via packages implementing reporting logic for the specific tools. Alternatively, third party tools may implement pull-based mechanisms to get information from the execution of a playbook via SOARCA.
```
Comment (RabbITCybErSeC) [inline docs/content/en/docs/core-components/reporting.md:78]: `third-party tools`

File: docs/content/en/docs/core-components/reporting.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+Reporter --left--> DB
+Reporter --right--> 3PTool
+
+```
+
+### Interfaces
+
+The logic and extensibility is implemented in the SOARCA architecture by means of reporting interfaces. At this stage, we implement an *IWorkflowReporter* to push information about the entire workflow to be executed, and an *IStepReporter* to push step-specific information as the steps of the workflow are executed.
+
+A high level *Reporter* class will implement both interfaces, and maintain the list of decomposer and executor reporters activated for the SOARCA instance. The *Reporter* class will invoke all reporting functions for each active reporter.
```
[the remark this answers is not in the document]
Comment (lucamrgs) [reply docs/content/en/docs/core-components/reporting.md:74]: Probably component is a better term for the architecture

File: docs/content/en/docs/core-components/reporting.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+---
+title: Reporting
+weight: 8
+categories: [architecture]
+tags: [apis]
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/core-components/reporting.md:5]: api

File: docs/content/en/docs/core-components/reporting.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
+---
+title: Reporting
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/core-components/reporting.md:2]: Maybe the title should be Reporter

File: docs/content/en/docs/core-components/reporting.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+### Interfaces
+
+The logic and extensibility is implemented in the SOARCA architecture by means of reporting interfaces. At this stage, we implement an *IWorkflowReporter* to push information about the entire workflow to be executed, and an *IStepReporter* to push step-specific information as the steps of the workflow are executed.
+
+A high level *Reporter* component will implement both interfaces, and maintain the list of decomposer and executor reporters activated for the SOARCA instance. The *Reporter* class will invoke all reporting functions for each active reporter.
+
+## Future plans
+
+At this stage, third-party tools integrations may be built in SOARCA via packages implementing reporting logic for the specific tools. Alternatively, third-party tools may implement pull-based mechanisms to get information from the execution of a playbook via SOARCA.
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/core-components/reporting.md:78]:
```suggestion
At this stage, third-party tools integrations may be built in SOARCA via packages implementing reporting logic for the specific tools. Alternatively, third-party tools may implement pull-based mechanisms (via the API) to get information from the execution of a playbook via SOARCA.
```

File: docs/content/en/docs/core-components/reporting.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+interface IStepReporter{
+    ReportStep(cacao.workflow.Step, cacao.Variables, error)
+}
+
+class Reporter {
+    stepReporters []IStepReporter
+    workflowReporters []IWorkflowReporter
+
+    registerStepReporter()
+    registerWorkflowReporter()
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/core-components/reporting.md:45]:
```suggestion
    RegisterStepReporter()
    RegisterWorkflowReporter()
```

File: docs/content/en/docs/core-components/reporting.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+}
+class DB
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/core-components/reporting.md:47]:
```suggestion
class Database as DB
```

File: docs/content/en/docs/core-components/reporting.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+---
+
+SOARCA utilizes push-based reporting to provide information on the instantiation of a CACAO workflow, and information on the execution of workflow steps.
+
+
+## General Reporting Architecture
+
+For the execution of a playbook, a *Decomposer* and invoked *Executor*s are injected with a *Reporter*. The *Reporter* maintains the reporting logic that reports execution information to a set of specified and available targets.
+
+A reporting target can be internal to SOARCA, such as the [database](https://cossas.github.io/SOARCA/docs/core-components/database/), and the [report] API. A reporting target can also be a third-party tool, such as an external SOAR, or incident case management system.
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/core-components/reporting.md:17]:
```suggestion
A reporting target can be internal to SOARCA, such as the [database](https://cossas.github.io/SOARCA/docs/core-components/database/), and the [report] API. A reporting target can also be a third-party tool, such as an external SOAR/ SIEM, or incident case management system.
```

**Review by MaartendeKruijf — COMMENTED** [review]
@lucamrgs looks nice some minor things

**Comment by RabbITCybErSeC** [comment]
@lucamrgs  ready for approval?

**Pull request merged**